File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/teacher/TeacherEntity.java
package kh.farrukh.sortmapperexamplejava.teacher;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import kh.farrukh.sortmapperexamplejava.common.audit.AuditableEntity;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Getter
@Setter
@Entity
@Table(name = "teachers")
@EntityListeners(AuditingEntityListener.class)
public class TeacherEntity extends AuditableEntity {

  private static final String TEACHER_ID_GENERATOR = "teacher_id_generator";
  private static final String TEACHER_ID_SEQUENCE = "teacher_id_sequence";

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = TEACHER_ID_GENERATOR)
  @SequenceGenerator(name = TEACHER_ID_GENERATOR, sequenceName = TEACHER_ID_SEQUENCE, allocationSize = 1)
  private Long id;

  @Column(name = "first_name", nullable = false, length = 50)
  private String firstName;

  @Column(name = "last_name", nullable = false, length = 50)
  private String lastName;

  @Column(name = "middle_name", length = 50)
  private String middleName;

  @Column(name = "age", nullable = false)
  private Integer age;

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/teacher/TeacherEntity.kt
package kh.farrukh.sortmapperexample.teacher

import jakarta.persistence.*
import kh.farrukh.sortmapperexample.common.audit.AuditableEntity
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@Table(name = "teachers")
@EntityListeners(AuditingEntityListener::class)
class TeacherEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = TEACHER_ID_GENERATOR)
    @SequenceGenerator(
        name = TEACHER_ID_GENERATOR,
        sequenceName = TEACHER_ID_SEQUENCE,
        allocationSize = 1
    )
    var id: Long? = null,
    @Column(name = "first_name", nullable = false, length = 50)
    var firstName: String,
    @Column(name = "last_name", nullable = false, length = 50)
    var lastName: String,
    @Column(name = "middle_name", length = 50)
    var middleName: String? = null,
    @Column(name = "age", nullable = false)
    var age: Int
) : AuditableEntity(
    createdAt = LocalDateTime.now(),
    updatedAt = LocalDateTime.now()
) {

    companion object {
        const val TEACHER_ID_GENERATOR = "teacher_id_generator"
        const val TEACHER_ID_SEQUENCE = "teacher_id_sequence"
    }

}


